src/lava/lib/optimization/problems/variables.py

```diff
@@ -93,7 +94,7 @@ def variable_set(self):
     @property
     def num_variables(self):
         """Number of variables in this set."""
-        return len(self._bounds)
+        return self._num_variables
```

Comment:
Why not 
`return len(self.domains) if self.domains is not None else None`
like in line 69?

Reply:
Because the data you get in the interface is different for continuous time and discrete problems. I didn't make any changes to the discrete interface